fvm/fvm.go

```diff
@@ -21,12 +21,24 @@ type Procedure interface {
 	Run(
 		ctx Context,
 		sth *state.StateHolder,
-		programs *programs.Programs,
+		programs *programs.TransactionPrograms,
 	) error
 
 	ComputationLimit(ctx Context) uint64
 	MemoryLimit(ctx Context) uint64
 	ShouldDisableMemoryAndInteractionLimits(ctx Context) bool
+
+	IsSnapshotReadTransaction() bool
+	IsNormalTransaction() bool
```

Comment:
nit: to me one single method returning an enum like `BootstrapProcedureType`, `ScriptProcedureType`, `TransactionProcedureType` is neater. but just my personal preference. not blocking.

Reply:
that's reasonable.  i'll make the change